components/browser/icons/src/main/java/mozilla/components/browser/icons/extension/IconMessage.kt

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.icons.extension
+
+import mozilla.components.browser.icons.IconRequest
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.lang.IllegalArgumentException
+
+private val typeMap: Map<String, IconRequest.Resource.Type> = mutableMapOf(
+    "icon" to IconRequest.Resource.Type.FAVICON,
+    "shortcut icon" to IconRequest.Resource.Type.FAVICON,
+    "fluid-icon" to IconRequest.Resource.Type.FLUID_ICON,
+    "apple-touch-icon" to IconRequest.Resource.Type.APPLE_TOUCH_ICON,
+    "image_src" to IconRequest.Resource.Type.IMAGE_SRC,
+    "apple-touch-icon image_src" to IconRequest.Resource.Type.APPLE_TOUCH_ICON,
+    "apple-touch-icon-precomposed" to IconRequest.Resource.Type.APPLE_TOUCH_ICON,
+    "og:image" to IconRequest.Resource.Type.OPENGRAPH,
+    "og:image:url" to IconRequest.Resource.Type.OPENGRAPH,
+    "og:image:secure_url" to IconRequest.Resource.Type.OPENGRAPH,
+    "twitter:image" to IconRequest.Resource.Type.TWITTER,
+    "msapplication-TileImage" to IconRequest.Resource.Type.MICROSOFT_TILE
+)
+
+private fun Map<String, IconRequest.Resource.Type>.reverseLookup(type: IconRequest.Resource.Type): String {
```

Comment:
Should we return a list here. I noticed on `typeMap` there are multiple keys for one given value, for instance:

`IconRequest.Resource.Type.OPENGRAPH` has ` ["og:image", "og:image:url", "og:image:secure_url"]`

`IconRequest.Resource.Type.APPLE_TOUCH_ICON` has `["apple-touch-icon","apple-touch-icon precomposed","apple-touch-icon image_src"]`

Reply:
The actual value used is not important in this case as long as when parsing it's deserialized to `IconRequest.Resource.Type.OPENGRAPH` again.

Reply:
👍